macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,278 @@
+meta:
+  id: mac_os_resource_snd
+  title: Classic MacOS Sound Resource
+  application: Sound Manager
+  endian: be
+  license: MIT
+  xref:
+    wikidata: Q7564684
+doc-ref: "https://developer.apple.com/library/archive/documentation/mac/pdf/Sound/Sound_Manager.pdf"
+doc: Sound resources were introduced in Classic MacOS with the Sound Manager program.
+  They can contain sound commands to generate sounds with given frequencies as well as sampled sound data.
+  They are mostly found in resource forks, but can occasionally appear standalone or embedded in other files.
+seq:
+  - id: format
+    type: u2
+  - id: num_data_formats
+    -orig-id: number_of_data_formats
+    type: u2
+    if: format == 1
+  - id: data_formats
+    type: data_format
+    repeat: expr
+    repeat-expr: num_data_formats
+    if: format == 1
+  - id: reference_count
+    type: u2
+    if: format == 2
+  - id: num_sound_commands
+    -orig-id: number_of_sound_commands
+    type: u2
+  - id: sound_commands
+    type: sound_command
+    repeat: expr
+    repeat-expr: num_sound_commands
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    seq:
+      - id: is_data_offset
+        type: b1
+      - id: cmd
+        type: b15
+        enum: cmd_type
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      sound_header_type:
+        pos: param2 + 20
+        type: u1
+        enum: sound_header_type
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+      sound_header:
+        pos: param2
+        size-eos: true
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+        type:
+          switch-on: sound_header_type
+          cases:
+            sound_header_type::standard: standard_sound_header
+            sound_header_type::extended: extended_sound_header
+            sound_header_type::compressed: compressed_sound_header
+  standard_sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or NIL if samples follow data structure)
+      - id: length
+        type: u4
+        doc: number of samples in array
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample’s encoding option
```

Comment:
I think there's a little point in using a Unicode symbol [_Right Single Quotation Mark_](https://unicode-table.com/en/2019/) `’` with code point `U+2019` instead of the normal [`U+0027` Apostrophe](https://unicode-table.com/en/0027/) `'` from the ASCII table.

It might look a bit nicer, but as the whole file is encoded in pure ASCII, sticking to ASCII chars will help with portability:

```suggestion
        doc: sample's encoding option
```